gapic/schema/wrappers.py

```diff
@@ -731,6 +749,17 @@ def names(self) -> FrozenSet[str]:
         # Done; return the answer.
         return frozenset(answer)
 
+    @utils.cached_property
+    def resource_messages(self) -> FrozenSet[MessageType]:
+        """Returns all the resource message types used in all
+        request fields in the service."""
+        return frozenset(
+            field.message
+            for method in self.methods.values()
+            for field in method.input.fields.values()
+            if field.message and field.message.resource_path
```

Comment:
You might have missed the file-level annotations (`google.api.resource_definition`).

Optional: consider filter out resources defined but not referenced. The monolith does that; If your goal is to maintain backward-compatibility it might be worth keeping the behavior. OTOH, the logic seems too implicit, I am also fine if you drop it until we found something broken :p

Reply:
Good catch! I did miss file-level annotations.
Where should those helper methods live? If they belong to the API but not to any particular service, there isn't an obvious client module where they should belong. Should they just be part of the base module? That would reduce duplication between multiple services that reference the same resource, but on the other hand the whole point of adding these methods back is backwards compatibility; if I'm reading the monolithic template correctly, all that's currently created are per-client-class helper methods.

Reply:
Yeah that's a good point. I would say check where the resource is referenced, but a quick search in googleapis doesn't give me any API that does so, and it's really hard to reason without one.

Now I realized that the five common resources should also be supported. In this case, the helper methods should be generated if they are referenced in any messages that are either the input, output or fields of input or output of any method in a service. Given this requirement, I think treating the file-level annotations the same way is probably the best solution.

Reply:
Okay, so file-level resources just get added to a client if the client refers to that resource in any way. I can live with that (and so will the users).
I think I may be missing something, though: the Python `google.api.resource_pb2` module doesn't have a `resource_definition` member.

Reply:
Oh, you probably need a new version of Python common protos. It's probably released manually? @lukesneeringer

Reply:
Are you okay with postponing the five common resources and file level resources? The first is already being tracked by #210, and I can file an issue for the second.

Reply:
SGTM